app/src/main/java/com/celzero/bravedns/ui/SettingsFragment.kt

```diff
@@ -614,26 +668,55 @@ class SettingsFragment : Fragment(R.layout.activity_settings_screen) {
+        val checkedItem = persistentState.notificationAction
+        alertBuilder.setSingleChoiceItems(items, checkedItem) { dialog, which ->
+            dialog.dismiss()
+            if (persistentState.notificationAction != which) {
+                when (which) {
+                    Constants.NOTIFICATION_ACTION_STOP -> {
+                        b.genSettingsNotificationDesc.text = getString(R.string.settings_notification_desc, getString(R.string.settings_notification_desc1))
+                        persistentState.notificationAction = Constants.NOTIFICATION_ACTION_STOP
+                    }
+                    Constants.NOTIFICATION_ACTION_DNS_FIREWALL -> {
+                        b.genSettingsNotificationDesc.text = getString(R.string.settings_notification_desc, getString(R.string.settings_notification_desc2))
+                        persistentState.notificationAction = Constants.NOTIFICATION_ACTION_DNS_FIREWALL
+                    }
+                    Constants.NOTIFICATION_ACTION_NONE -> {
+                        b.genSettingsNotificationDesc.text = getString(R.string.settings_notification_desc, getString(R.string.settings_notification_desc3))
+                        persistentState.notificationAction = Constants.NOTIFICATION_ACTION_NONE
+                    }
+                }
+            }
+        }
+        // Create the AlertDialog
+        val alertDialog: AlertDialog = alertBuilder.create()
+        // Set other dialog properties
+        alertDialog.show()
+    }
+
     override fun onResume() {
         super.onResume()
         detectLockDownMode()
         if (!persistentState.localBlocklistEnabled) {
             b.settingsActivityOnDeviceBlockDesc.text = getString(R.string.settings_local_blockList_desc1)
-        }else {
+        } else {
             val count = persistentState.numberOfLocalBlocklists
-            if (persistentState.localBlocklistEnabled && persistentState.blockListFilesDownloaded && persistentState.getLocalBlockListStamp().isNotEmpty()) {
-                if(count !=0 ){
+            if (persistentState.localBlocklistEnabled && persistentState.blockListFilesDownloaded && persistentState.localBlockListStamp.isNotEmpty()) {
+                if (count != 0) {
                     b.settingsActivityOnDeviceBlockDesc.text = getString(R.string.settings_local_blocklist_in_use, count.toString())
-                }else {
+                } else {
                     b.settingsActivityOnDeviceBlockDesc.text = getString(R.string.settings_local_blocklist_desc6)
                 }
-            } else if(count != 0){
+            } else if (count != 0) {
                 b.settingsActivityOnDeviceBlockDesc.text = getString(R.string.settings_local_blocklist_in_use, count.toString())
-            }else {
+            } else {
```

Comment:
Move this to its own method: `handleOnDeviceBlocklist` or something better.

also, simplify the code block, using a guard:

```
// verify if the following guard statement is correct
if (!persistentState.localBlocklistEnabled || !persistentState.blockListFilesDownloaded || persistentState.localBlockListStamp.isNotEmpty()) {
  b.settingsActivityOnDeviceBlockDesc.text = getString(R.string.settings_local_blockList_desc1)
  return
}

val count = persistentState.numberOfLocalBlocklists
if (count > 0) {
  b.settingsActivityOnDeviceBlockDesc.text = getString(R.string.settings_local_blocklist_in_use, count.toString())
} else {
  b.settingsActivityOnDeviceBlockDesc.text = getString(R.string.settings_local_blocklist_desc6)
}
```

Reply:
Moved to `handleOnDeviceBlocklist` and simplified the code block.